esx_service/vmdk_ops.py

```diff
@@ -823,25 +840,57 @@ def get_datastore_name(datastore_url):
 
     return datastore_name
 
-def authorize_check(vm_uuid, datastore_url, cmd, opts, use_default_ds, datastore, vm_datastore):
+def authorize_check(vm_uuid, datastore_url, cmd, opts, use_default_ds, datastore, vm_datastore_url, vm_datastore):
```

Comment:
To do an authorize check, we need to pass 5 datastore related parameters. This is an indication that something is wrong either in our implementation or in our design. I don't have good suggestions right now as the existing code are just too convoluted. It seems not straightforward to improve the quality without a significant refactoring (or design change).